migrations/1_create_users_projects_files.js

```diff
@@ -1,84 +1,54 @@
-    t.integer('published_id').notNullable().references('id').inTable('publishedProjects').onDelete('CASCADE');
-    t.integer('file_id').references('id').inTable('files');
-    t.text('path').notNullable();
-    t.specificType('buffer', 'bytea').notNullable();
-  });
-};
-
-
-exports.down = function (knex) {
+exports.up = function (knex, Promise) {
   return Promise.resolve()
   .then(function() {
-    return knex.schema.hasTable('users')
-    .then(function(exists) {
-      if (exists) {
-        return knex.raw('DROP TABLE users CASCADE');
-      }
+    return knex.schema.createTable('publishedProjects', function(t) {
+      t.increments('id');
+      t.text('title').notNullable();
+      t.text('tags');
+      t.text('description');
     });
   })
   .then(function() {
-    return knex.schema.hasTable('projects')
-    .then(function(exists) {
-      if (exists) {
-        return knex.raw('DROP TABLE projects CASCADE');
-      }
+    return knex.schema.createTable('users', function (t) {
+      t.increments('id');
+      t.text('name').notNullable().unique();
     });
   })
   .then(function() {
-    return knex.schema.hasTable('files')
-    .then(function(exists) {
-      if (exists) {
-        return knex.raw('DROP TABLE files CASCADE');
-      }
+    return knex.schema.createTable('projects', function (t) {
+      t.increments('id');
+      t.integer('user_id').notNullable().references('id').inTable('users').onDelete('CASCADE');
+      t.integer('published_id').references('id').inTable('publishedProjects');
+      t.text('title').notNullable();
+      t.text('tags');
+      t.text('description');
+      t.text('publish_url');
+      t.text('date_created').notNullable();
+      t.text('date_updated').notNullable();
     });
   })
   .then(function() {
-    return knex.schema.hasTable('publishedProjects')
-    .then(function(exists) {
-      if (exists) {
-        return knex.raw('DROP TABLE "publishedProjects" CASCADE');
-      }
+    return knex.schema.createTable('files', function (t) {
+      t.increments('id');
+      t.integer('project_id').notNullable().references('id').inTable('projects').onDelete('CASCADE');
+      t.text('path').notNullable();
+      t.specificType('buffer', 'bytea').notNullable();
     });
   })
   .then(function() {
-    return knex.schema.hasTable('publishedFiles')
-    .then(function(exists) {
-      if (exists) {
-        return knex.raw('DROP TABLE "publishedFiles"');
-      }
+    return knex.schema.createTable('publishedFiles', function(t) {
+      t.increments('id');
+      t.integer('published_id').notNullable().references('id').inTable('publishedProjects').onDelete('CASCADE');
+      t.integer('file_id').references('id').inTable('files');
+      t.text('path').notNullable();
+      t.specificType('buffer', 'bytea').notNullable();
     });
   });
 };
+
+
+exports.down = function (knex, Promise) {
+  // To prevent migrating down to "there aren't even any tables",
+  // there is no down() run.
```

Comment:
we *literally* deleted the production database because the initial migration is allowed to wipe the database. We're not doing that anymore.